staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -440,6 +440,11 @@ func (w *watchCache) List() []interface{} {
 // You HAVE TO explicitly call w.RUnlock() after this function.
 func (w *watchCache) waitUntilFreshAndBlock(ctx context.Context, resourceVersion uint64) error {
 	startTime := w.clock.Now()
+	defer func() {
+		if resourceVersion > 0 {
+			metrics.WatchCacheReadWait.WithContext(ctx).WithLabelValues(w.groupResource.String()).Observe(w.clock.Since(startTime).Seconds())
```

Comment:
Wondering whether it would make sense to record the metric only when `w.resourceVersion < resourceVersion ` holds true? Otherwise, it means we actually didn't wait and the cache was fresh, right?